isisd/isis_pdu.c

```diff
@@ -582,21 +582,37 @@ static int process_hello(uint8_t pdu_type, struct isis_circuit *circuit,
 		if (circuit->circ_type != CIRCUIT_T_P2P) {
 			zlog_warn("p2p hello on non p2p circuit");
 			circuit->rej_adjacencies++;
+			if (circuit->is_type == IS_LEVEL_1) {
+				circuit->area->rej_adjacencies[0]++;
+			} else if (circuit->is_type == IS_LEVEL_2) {
+				circuit->area->rej_adjacencies[1]++;
+			} else {
+				circuit->area->rej_adjacencies[0]++;
+				circuit->area->rej_adjacencies[1]++;
+			}
```

Comment:
May be a macro or function would be useful here as you repeat this portion of code at many place below

Reply:
added a subroutine.

In the test setup there is ecmp route for 25.0.0.0/24 from R1's point of view,  but 
there are no segment-routing or ecmp leafs in RFC 4444 read-only mib tables.